base/cvd/cuttlefish/host/commands/cvd/server_command/create.cpp

```diff
@@ -0,0 +1,339 @@
+}
+
+std::string DefaultProductPath(const cvd_common::Envs& envs) {
+  if (Contains(envs, kAndroidProductOut)) {
+    return envs.at(kAndroidProductOut);
+  }
+  if (Contains(envs, "HOME")) {
+    return envs.at("HOME");
+  }
+  return CurrentDirectory();
+}
+
+struct CreateFlags {
+  std::string host_path;
+  std::string product_path;
+  bool start;
+  std::string config_file;
+};
+Result<CreateFlags> ParseCommandFlags(const cvd_common::Envs& envs,
+                                      cvd_common::Args& args) {
+  CreateFlags flag_values{
+      .host_path = DefaultHostPath(envs),
+      .product_path = DefaultProductPath(envs),
+      .start = true,
+      .config_file = "",
+  };
+  std::vector<Flag> flags = {
+      GflagsCompatFlag("host_path", flag_values.host_path),
+      GflagsCompatFlag("product_path", flag_values.product_path),
+      GflagsCompatFlag("start", flag_values.start),
+      GflagsCompatFlag("config_file", flag_values.config_file),
+  };
+  CF_EXPECT(ConsumeFlags(flags, args));
+  return flag_values;
+}
+
+RequestWithStdio CreateLoadCommand(const RequestWithStdio& request,
+                                   cvd_common::Args& args,
+                                   const std::string& config_file) {
+  cvd::Request request_proto;
+  auto& load_command = *request_proto.mutable_command_request();
+  *load_command.mutable_env() = request.Message().command_request().env();
+  load_command.set_working_directory(
+      request.Message().command_request().working_directory());
+  load_command.add_args("cvd");
+  load_command.add_args("load");
+  for (const auto& arg : args) {
+    load_command.add_args(arg);
+  }
+  load_command.add_args(config_file);
+  return RequestWithStdio(request_proto, request.FileDescriptors());
+}
+
+RequestWithStdio CreateStartCommand(const RequestWithStdio& request,
+                                    const selector::LocalInstanceGroup& group,
+                                    const cvd_common::Args& args,
+                                    const cvd_common::Envs& envs) {
+  cvd::Request request_proto;
+  auto& load_command = *request_proto.mutable_command_request();
+  for (const auto& [key, value] : envs) {
+    (*load_command.mutable_env())[key] = value;
+  }
+  load_command.set_working_directory(
+      request.Message().command_request().working_directory());
+  load_command.mutable_selector_opts()->clear_args();
+  load_command.mutable_selector_opts()->add_args("--group_name");
+  load_command.mutable_selector_opts()->add_args(group.GroupName());
+  load_command.add_args("cvd");
+  load_command.add_args("start");
+  for (const auto& arg : args) {
+    load_command.add_args(arg);
+  }
+  return RequestWithStdio(request_proto, request.FileDescriptors());
+}
+
+Result<cvd_common::Envs> GetEnvs(const RequestWithStdio& request) {
+  cvd_common::Envs envs =
+      cvd_common::ConvertToEnvs(request.Message().command_request().env());
+  if (envs.at("HOME").empty()) {
```

Comment:
`at` throws a `std::out_of_range` exception on access failure and this happens before the `Contains` check, so it could crash in an unexpected way.